File: backend/src/settings/lrembecki.core.settings/ViewModels/EmailVM.cs
using lrembecki.core.settings.Entities;

namespace lrembecki.core.settings.ViewModels;

public record EmailVM(
    Guid Id,
    bool IsActive,
    string SmtpServer,
    int SmtpPort,
    string SmtpUsername,
    string SmtpPassword,
    string FromAddress,
    string FromName,
    string ReplyToAddress,
    string ReplyToName)
{
    public static EmailVM Map(EmailEntity email)
    {
        if (email == null) return null!;

        return new(
            Id: email.Id,
            IsActive: email.IsActive,
            SmtpServer: email.SmtpServer,
            SmtpPort: email.SmtpPort,
            SmtpUsername: email.SmtpUsername,
            SmtpPassword: email.SmtpPassword,
            FromAddress: email.FromAddress,
            FromName: email.FromName,
            ReplyToAddress: email.ReplyToAddress,
            ReplyToName: email.ReplyToName
        );
    }
}

File: frontend/src/app/modules/settings/emails/email.vm.ts
export class EmailVM {
  id: string = null!;
  isActive: boolean = false;
  smtpServer: string = '';
  smtpPort: number = 0;
  smtpUsername: string = '';
  smtpPassword: string = '';
  fromAddress: string = '';
  fromName: string = '';
  replyToAddress: string = '';
  replyToName: string = '';
  constructor(init?: Partial<EmailVM>) {
    Object.assign(this, init);
  }
}
